assembl/static/js/app/views/message.js

```diff
@@ -1524,6 +1546,52 @@ var MessageView = Marionette.LayoutView.extend({
     this.listenToOnce(this.messageModerationOptionsView, 'moderationOptionsClose', this.onModerationOptionsClose);
   },
 
+  onDeleteMessageClick: function(ev){
+    var that = this;
+    // We could try to minimize context switching for the user, by scrolling the viewport to the message the user wants to delete, as soon as the confirmation popin opens, using this line of code:
+    // that.messageListView.scrollToMessage(that.model, false, false);
+    
+    var onSubmit = function(ev){
+      var analytics = Analytics.getInstance();
+      analytics.trackEvent(analytics.events.MESSAGE_LIKED);
+      // we could use that.model.destroy() instead, and add || method == "delete" to models/message.js::sync()
+      var message_delete_url = that.model.getApiV2Url();
+      Promise.resolve(
+        $.ajax(message_delete_url, {
```

Comment:
I wrote it this way first and we thought about the other way togeter afterwards, both should work I guess, but I thought there is a small risk that this becomes a bit tricky and produce unanticipated effects. So as we are a bit in a hurry, I prefered to keep it as is rather than redo full AQ. Let's talk about it together when we have a moment!